actgen/utils.py

```diff
@@ -4,12 +4,104 @@
+                            help='Path to hyperparameters csv file')
+        parser.add_argument('--test', default=False, action='store_true',
+                            help='Enable test mode for quickly checking configuration works')
+        parser.add_argument('--results_dir', type=str, default='./results/',
+                            help='Path to the result directory to save model files')
+        parser.add_argument('--tag', type=str, default='default_exp',
+                            help='A tag for the current experiment, used as a subdirectory name for saving models')
+        parser.add_argument('--disable_gpu', default=False, action='store_true',
+                            help='enforce training on CPU')
+        return parser
+
+    def parse_unknown_args(self, parser):
+        args, unknown = parser.parse_known_args()
+        other_args = {
+            (remove_prefix(key, '--'), val)
+            for (key, val) in zip(unknown[::2], unknown[1::2])
+        }
+        args.other_args = other_args
+        return args
+    
+    def load_hyperparams(self, args):
+        params = load_hyperparams(args.hyperparams)
+        for arg_name, arg_value in vars(args).items():
+            if arg_name == 'hyperparams':
+                continue
+            params[arg_name] = arg_value
+        for arg_name, arg_value in args.other_args:
+            update_param(params, arg_name, arg_value)
+        return params
+    
+    def make_gym_env(self):
+        if self.params['atari']:
+            env = wrap.make_deepmind_atari(
+                    self.params['env_name'],
+                    max_episode_steps=(
+                    self.params['max_episode_steps'] if self.params['max_episode_steps'] > 0 else None
+                    ),
+                    episode_life=self.params['episode_life'],
+                    clip_rewards=self.params['clip_rewards'],
+                    frame_stack=self.params['frame_stack'],
+                    scale=self.params['scale_pixel_values'])
+        else:
+            assert self.params['env_name'] in ['CartPole-v0', 'Pendulum-v0', 'LunarLander-v2']
+            env = gym.make(self.params['env_name'])
+            env = wrap.FixedDurationHack(env)
+        if isinstance(env.action_space, gym.spaces.Box):
+            env = wrap.DiscreteBox(env)
+        if self.params['random_actions']:
+            logging.info('making all duplicate actions random actions')
+            env = wrap.RandomActions(env, self.params['duplicate'])
+        else:
+            logging.info('making {} sets of exactly same duplicate actions'.format(self.params['duplicate']))
+            env = wrap.DuplicateActions(env, self.params['duplicate'])
+        env = wrap.TorchInterface(env)
+        return env
+    
+    def determine_device(self):
+        self.params['device'] = determine_device(self.params['disable_gpu'])
+        logging.info('training on device {}'.format(self.params['device']))
```

Comment:
I don't love that `determine_device` is being overloaded here. Why not just move the `logging.info` into the global function and get rid of this method?